File: Python/20. Valid Parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        
        original = ""
        while s != original:
            original = s
            s = s.replace("()","")
            s = s.replace("[]","")
            s = s.replace("{}","")
        if s == "":
            return True
        else:
            return False
test = Solution()
print(test.isValid("()()()[]{(}"))

File: CPP/20. Valid Parentheses.cpp
using namespace std;
#include <iostream>
#include <vector>
class Solution {
public:
    bool isValid(string s) {
        vector<char> stack;
        for (int i = 0; i < s.length(); i++){
            if (s[i] == '(' || s[i] == '[' || s[i] == '{'){
                stack.push_back(s[i]);
                // cout << s[i];
            } else if (s[i] == ')' && stack.size()>0 && stack.back() != '('){
                return false;
            } else if (s[i] == ']' && stack.size()>0 && stack.back() != '['){
                return false;
            } else if (s[i] == '}' && stack.size()>0 && stack.back() != '{'){
                return false;
            } else if (stack.size() == 0){
                return false;
            } else {
                // cout << s[i];
                stack.pop_back();
                // cout << stack.size();
            }            
        }
        if (stack.size() > 0){
            return false;
        }
        return true;
    }
};